plugins/inputs/elasticsearch_query/elasticsearch_query.go

```diff
@@ -0,0 +1,287 @@
+  ## Setting to "0s" will disable the health check (not recommended in production)
+  health_check_interval = "10s"
+  ## HTTP basic authentication details (eg. when using x-pack)
+  # username = "telegraf"
+  # password = "mypassword"
+
+  ## Optional SSL Config
+  # ssl_ca = "/etc/telegraf/ca.pem"
+  # ssl_cert = "/etc/telegraf/cert.pem"
+  # ssl_key = "/etc/telegraf/key.pem"
+  ## Use SSL but skip chain & host verification
+  # insecure_skip_verify = false
+
+  ## Examples:
+
+  # Search the average response time, per URI and per response status code
+[[inputs.elasticsearch_query.aggregation]]
+  measurement_name = "http_logs" 	# Name of destination measurement
+  index = "my-index-*" 				# Elasticsearch index to query
+  filter_query = "*" 				# Optional: Lucene query to filter results
+  metric_fields = ["response_time"] # Optional: field to aggregate values (must be numeric fields, for calculation)
+  metric_function = "avg" 			# Optional: function to use on aggregation
+  tags = ["URI.keyword", "response.keyword"] # Optional: fields to be used as tags (must be non-analyzed fields, aggregations will be performed per tag)
+  include_missing_tag = true 		# Optional: set to true to not ignore documents where the tag(s) specified above does not exist
+  missing_tag_value = "null" 		# Optional: value of the tag set for documents where the tag does not exist
+  date_field = "@timestamp" 		# Timestamp field, mandatory
+  query_period = "1m"  				# Time window to query (eg. "1m" to query documents from last minute). Normally should be set to same as collection interval
+ 
+# Search the maximum response time per method and per URI
+[[inputs.elasticsearch_query.aggregation]]
+  measurement_name = "http_logs"
+  index = "my-index-*"
+  filter_query = "*"
+  metric_fields = ["response_time"]
+  metric_function = "max"
+  tags = ["method.keyword","URI.keyword"]
+  include_missing_tag = false
+  missing_tag_value = "null"
+  date_field = "@timestamp"
+  query_period = "1m"
+
+# Search number of documents matching a filter query
+[[inputs.elasticsearch_query.aggregation]]
+  measurement_name = "http_logs"
+  index = "*"
+  filter_query = "product_1 AND HEAD"
+  query_period = "1m"
+  date_field = "@timestamp"
+
+# Search number of documents matching a filter query per response status code
+[[inputs.elasticsearch_query.aggregation]]
+  measurement_name = "http_logs"
+  index = "*"
+  filter_query = "downloads"
+  tags = ["response.keyword"]
+  include_missing_tag = false
+  date_field = "@timestamp"
+  query_period = "1m"
+`
+
+type ElasticsearchQuery struct {
+	URLs                []string `toml:"urls"`
+	Username            string
+	Password            string
+	EnableSniffer       bool
+	Tracelog            bool
+	Timeout             internal.Duration
+	HealthCheckInterval internal.Duration
+	Aggregations        []Aggregation `toml:"aggregation"`
+	SSLCA               string        `toml:"ssl_ca"`   // Path to CA file
```

Comment:
No need for these SSL/TLS attributes if you use `./internal/tls` and the shared tls config [like apache uses](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/apache/apache.go#L96)